### PR TITLE
OCPBUGS-2429: Fix TestReloadInterval min value test cases and race conditions

### PR DESCRIPTION
**Pull request by gcs278**
OCPBUGS-2429: Fix TestReloadInterval min value test cases to 1s and fix race condition in which it passes as a false positive by forcing the ingress operator to recreated the router deployment instead of waiting on the ingress operator to report Available.

- `test/e2e/reloadinterval_test.go`: Add deletion of router deployment to trigger reconcile and fix min value test cases

### CODE REVIEW

**Comment by Miciah** [comment]
/retest-required
now that #839 has merged.

File: test/e2e/reloadinterval_test.go [hunk 1]
```diff
@@ -73,9 +73,9 @@ func TestReloadInterval(t *testing.T) {
 
 		// set ReloadInterval to a value outside allowed bounds
 		// expect no error
-		{"set reloadInterval 200ns", metav1.Duration{Duration: 200 * time.Nanosecond}, "5s", true},
-		{"set reloadInterval 5us", metav1.Duration{Duration: 5 * time.Microsecond}, "5s", true},
-		{"set reloadInterval 500ms", metav1.Duration{Duration: 500 * time.Millisecond}, "5s", true},
+		{"set reloadInterval 200ns", metav1.Duration{Duration: 200 * time.Nanosecond}, "1s", true},
+		{"set reloadInterval 5us", metav1.Duration{Duration: 5 * time.Microsecond}, "1s", true},
```
Comment (miheer) [inline test/e2e/reloadinterval_test.go:77]: does setting to 1s  really help ? Won't  it cause load on the router ?

**Comment by miheer** [comment]
The PR lgtm so /lgtm

File: test/e2e/reloadinterval_test.go (hunk 1 above)
Comment (gcs278) [reply test/e2e/reloadinterval_test.go:77]: Not exactly - it's the value the test is looking for in the router deployment ENV variable. So no load affects, just a comparison value that was wrong.

**Comment by ShudiLi** [comment]
cancel /label qe-approve

**Pull request merged**